sympy/solvers/tests/test_solveset.py

```diff
@@ -551,19 +555,34 @@ def test_poly_gens():
     assert solveset_real(4**(2*(x**2) + 2*x) - 8, x) == \
         FiniteSet(-Rational(3, 2), S.Half)
 
+def test_expo():
+    x = Symbol('x')
+    assert solveset(4**(5-9*x) - 8**(2-x), x, S.Reals) == FiniteSet(Rational(4, 15))
```

Comment:
Well, as of now I haven't made it to work for complex domain. I will be making few changes in `_invert_complex` and then add the test cases.